op-service/txmgr/txmgr.go

```diff
@@ -74,12 +87,55 @@ type SimpleTxManager struct {
 	Config // embed the config directly
 	name   string
 
-	backend ReceiptSource
+	backend L1Connection
 	l       log.Logger
 }
 
+// IncreaseGasPrice takes the previous transaction & potentially clones then signs it with a higher tip.
+// If the basefee + priority fee did not increase by a minimum percent (geth's replacement percent) an
+// error will be returned.
+// We do not re-estimate the amount of gas used because for some stateful transactions (like output proposals) the
+// act of including the transaction renders the repeat of the transaction invalid.
+func (m *SimpleTxManager) IncreaseGasPrice(ctx context.Context, tx *types.Transaction) (*types.Transaction, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	var gasTipCap, gasFeeCap *big.Int
+
+	if tip, err := m.backend.SuggestGasTipCap(ctx); err != nil {
```

Comment:
Are we sure all clients support this RPC?

Reply:
I'm reasonably sure. I can check RPC provider docs though

Reply:
Ok, here's the support matrix. FYI internally we previously used this RPC method

eth_FeeHistory: geth, alchemy, infura
eth_maxPriorityFee (SuggestGasTip): geth, alchemy
quicknode doesn't seem to support either based on my reading of it.

Reply:
we shouldn't err in the case when the rpc is not present

Reply:
Ideally yes. I just realized that the place we will need to fix this is in `SuggestGasTipCap` or whatever the interface we define is.

Reply:
Going to resolve this convo because it is not a regression, but I have created a ticket for it at https://linear.app/optimism/issue/CLI-3318/support-multiple-l1-rpc-providers-with-the-ethconnection-client-in-the